services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowPriorityRerouteSpec.groovy

```diff
@@ -0,0 +1,129 @@
+        }
+
+        and: "Reroute procedure was done based on the priority field"
+        flows.sort { it.priority }*.id == northbound.getAllFlows().sort { it.lastUpdated }*.id
+
+        and: "Cleanup: revert system to original state"
+        northbound.portUp(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
+        flows.each { flowHelper.deleteFlow(it.id) }
+        Wrappers.wait(WAIT_OFFSET + discoveryInterval) {
+            assert islUtils.getIslInfo(islToBreak).get().state == IslChangeType.DISCOVERED
+        }
+        database.resetCosts()
+    }
+
+    @Ignore("https://github.com/telstra/open-kilda/issues/2211")
+    def "System is able to reroute(intentional) flow in the correct order based on the priority field"() {
+        given: "3 flows on the same path, with alt paths available"
+        def switches = topology.activeSwitches
+        List<List<PathNode>> allPaths = []
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.unique { it.sort() }.find { Switch src, Switch dst ->
+            allPaths = database.getPaths(src.dpId, dst.dpId)*.path
+            allPaths.size() > 2
+        }
+        List<FlowPayload> flows = []
+
+        def newPriority = 300
+
+        3.times {
+            def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+            flow.maximumBandwidth = 10000
+            flow.priority = newPriority
+            flowHelper.addFlow(flow)
+            newPriority -= 100
+            flows << flow
+        }
+        def currentPath = pathHelper.convert(northbound.getFlowPath(flows[0].id))
+        //ensure all flows are on the same path
+        assert pathHelper.convert(northbound.getFlowPath(flows[1].id)) == currentPath
+        assert pathHelper.convert(northbound.getFlowPath(flows[2].id)) == currentPath
+
+        when: "Make another path more preferable"
+        def newPath = allPaths.find { it != currentPath }
+        allPaths.findAll { it != newPath }.each { pathHelper.makePathMorePreferable(newPath, it) }
+
+        and: "Init simultaneous reroute for all flows"
+        def isl = pathHelper.getInvolvedIsls(currentPath).first()
+        northbound.rerouteLinkFlows(isl.srcSwitch.dpId, isl.srcPort, isl.dstSwitch.dpId, isl.dstPort)
+
+        then: "Flows were rerouted"
+        Wrappers.wait(rerouteDelay + WAIT_OFFSET) {
+            assert pathHelper.convert(northbound.getFlowPath(flows[0].id)) == newPath
+            assert pathHelper.convert(northbound.getFlowPath(flows[1].id)) == newPath
+            assert pathHelper.convert(northbound.getFlowPath(flows[2].id)) == newPath
+        }
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getFlowStatus(flows[0].id).status == FlowState.UP
+            assert northbound.getFlowStatus(flows[1].id).status == FlowState.UP
+            assert northbound.getFlowStatus(flows[2].id).status == FlowState.UP
+        }
```

Comment:
I'd rather you simplified these two blocks:
```
Wrappers.wait(WAIT_OFFSET) {
   flows.each {
       assert pathHelper.convert(northbound.getFlowPath(it.id)) == newPath
       assert northbound.getFlowStatus(it.id).status == FlowState.UP
   }   
}
```

Reply:
Fixed